### PR TITLE
LPS-158296 Create component for Length Field

### PR DESCRIPTION
**Pull request by veroglez**

### CODE REVIEW

**Comment by liferay-echo** [comment]
Thank you for submitting this PR. 👏

Make sure it follows the [Echo Pull Request Template](https://gist.github.com/liferay-echo/5a4455285ff1800f9da146816d85e67d#file-pull-request-template-md) if you haven't already done so. 

Learn more about it [here](https://liferay.atlassian.net/wiki/spaces/ENGECHO/pages/2027455014/Echo+Pull+Request+Etiquette).

**Comment by veroglez** [comment]
ci:test:sf

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/app/components/fragment-configuration-fields/LengthField.js [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+												onValueSelect(
+													field.name,
+													valueWithUnits
+												);
+											}
+										}
+									}}
+								>
+									{unit.toUpperCase()}
+								</ClayDropDown.Item>
```
Comment (p2kmgcl) [inline modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/app/components/fragment-configuration-fields/LengthField.js:118]: Since this is a fixed list, we can try to render it as a separated component (passing onClick handler) so we do not need to compute all list item for every render

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/app/components/fragment-configuration-fields/LengthField.js [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+					<ClayDropDown
+						active={active}
+						alignmentPosition={Align.BottomRight}
+						aria-labelledby={dropdownId}
+						menuElementAttrs={{
+							className: 'page-editor__length-field__dropdown',
+							containerProps: {
+								className: 'cadmin',
+							},
+						}}
```
[the remark this answers is not in the document]
Comment (veroglez) [reply modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/app/components/fragment-configuration-fields/LengthField.js]: Yes, you're right! here we can discuss a series of ideas that I have in mind

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/app/components/fragment-configuration-fields/LengthField.js [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+					>
+						<ClayDropDown.ItemList>
+							{UNITS.map((unit) => (
+								<ClayDropDown.Item
+									key={unit}
+									onClick={() => {
+										setActive(false);
+										setNextUnit(unit);
+
+										if (Number(nextValue)) {
```
[the remark this answers is not in the document]
Comment (veroglez) [reply modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/app/components/fragment-configuration-fields/LengthField.js]: At this moment the input is type number, so the nextValue can't be something different from a number. That will be part of the following task where the `Custom` option will be implemented.

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/app/components/fragment-configuration-fields/LengthField.js [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+import {useId} from '../../utils/useId';
+
+const KEYS_NOT_ALLOWED = ['+', ',', 'e'];
+
+const REGEX = /[-(0-9).]+|[a-zA-Z]+|%/g;
+
+const UNITS = ['px', '%', 'em', 'rem', 'vw', 'vh'];
+
+const isNumber = (value) => !isNaN(parseFloat(value));
+export function LengthField({field, onValueSelect, value}) {
```
Comment (p2kmgcl) [inline modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/app/components/fragment-configuration-fields/LengthField.js:32]: Please send another PR with a line break between these two or my head will implode :crab:

**Comment by p2kmgcl** [comment]
ci:forward
